src/convertToObject.js

```diff
@@ -6,7 +6,19 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  if (!sourceString.trim()) {
+    return {};
+  }
+
+  return sourceString
+    .split(';')
+    .map((keyValue) => keyValue.split(':').map((str) => str.trim()))
+    .filter(([key, value]) => key && value)
+    .reduce((accumulator, [key, value]) => {
```

Comment:
The 'reduce' method is used appropriately to construct the final object from the array of key-value pairs. This is a good use of array methods to avoid manual loops.